### PR TITLE
feat(eks): support for installing default networking add-ons

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #30804.

### Description of changes

- Add `bootstrapSelfManagedAddons` prop to `ClusterProps`

### Description of how you validated changes

Add both unit and integ tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by sumupitchayan — COMMENTED** [review]
Thanks for the PR @badmintoncryer  - you mention adding unit tests in the PR description but I don't see them here?

**Comment by badmintoncryer** [comment]
@sumupitchayan Thanks! I've completely forgotten to add unit test. I've added it.

File: packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-bootstrap-self-managed-addons.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class EksClusterStack extends Stack {
+  constructor(scope: App, id: string) {
+    super(scope, id);
+
+    const vpc = new ec2.Vpc(this, 'Vpc', { natGateways: 1 });
+    new eks.Cluster(this, 'EksCluster', {
+      vpc,
+      ...getClusterVersionConfig(this, eks.KubernetesVersion.V1_30),
+      bootstrapSelfManagedAddons: true,
```
Comment (sumupitchayan) [inline packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-bootstrap-self-managed-addons.ts:15]: Thanks for the PR @badmintoncryer - I see from the [CFN docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-cluster.html#cfn-eks-cluster-bootstrapselfmanagedaddons) that by default, the networking addons include vpc-cni, coredns, and kube-proxy. Maybe instead the integ test should pass in `bootstrapSelfManagedAddons: false` and assert somehow that the addons were not installed?

**Comment by badmintoncryer** [comment]
@sumupitchayan Apologize for the delay.

> Maybe instead the integ test should pass in bootstrapSelfManagedAddons: false and assert somehow that the addons were not installed?

I tried to [add integ test](packages/@aws-cdk-testing/framework-integ/test/aws-eks/test/integ.eks-bootstrap-self-managed-addons.ts) and noticed that creating an EKS cluster does not install any add-ons whether or not bootstrapSelfManagedAddons is enabled or not.

|bootstrapSelfManagedAddons|installed add-ons|
|-|-|
|true|nothing|
|false|nothing|
|undefined|nothing|

[Official document](https://docs.aws.amazon.com/eks/latest/userguide/eks-add-ons.html) says that default add-ons are always installed for the EKS cluster.

However, is my understanding correct that when creating an EKS cluster via CDK (eks.Cluster construct), no add-ons are installed?

**Pull request closed without merging**